plugin/restore/mysql/mysql.go

```diff
@@ -680,6 +680,189 @@ func (r *Restore) GetSortedBinlogFilesMetaOnServer(ctx context.Context) ([]Binlo
 	return sortBinlogFiles(binlogFiles), nil
 }
 
+// GetBinlogCoordinateByTs converts a timestamp to binlog coordinate using local binlog files.
+func (r *Restore) GetBinlogCoordinateByTs(ctx context.Context, targetTs int64) (*api.BinlogInfo, error) {
+	binlogFilesLocalSorted, err := GetSortedLocalBinlogFiles(r.binlogDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read sorted local binlog files, error: %w", err)
+	}
+	if len(binlogFilesLocalSorted) == 0 {
+		return nil, fmt.Errorf("no local binlog files found")
+	}
+	if !binlogFilesAreContinuous(binlogFilesLocalSorted) {
+		return nil, fmt.Errorf("local binlog files are not continuous")
+	}
+
+	var binlogFileTarget BinlogFile
```

Comment:
In this case, I might prefer "var binlogFileTarget *BinlogFile" to save foundBinlogFile. Same below.

Reply:
SG.